tests/rptest/services/producer_swarm.py

```diff
@@ -124,7 +131,30 @@ def start_node(self, node, clean=None):
             f"producer_swarm metrics endpoint at {self._remote_url(node, 'metrics/summary')} failed to answer after {30} sec",
         )
 
-        self._node = node
+    def wait_for_all_started(self):
+        """Wait until the requested number of producers have started. Note that if the expected
+        swarm runtime (messages / rate) is short, this may fail as all producers and start and
```

Comment:
> this shouldn't happen because we set the message count 

In `test_max_connections`? Yes it definitely shouldn't happen there, this was a comment for other users who might pass in 10 seconds worth of messages or something.

Reply:
> And do it below as well actually?

Do what below? I do take the startup pacing into account (it's more or less the only thing taken into account).

Reply:
> Do what below?

Sorry missed a word. Wanted to say "we do it below as well" so what I mean is that we also take it into account in this function but like you say that doesn't help if the actual starting code doesn't do the same.